dashboard/client/api/kube-api.ts

```diff
@@ -32,20 +32,31 @@ export interface IKubeApiLinkRef {
   namespace?: string;
 }
 
+export interface IKubeApiLinkBase extends IKubeApiLinkRef {
+  apiBase: string;
+  apiGroup: string;
+  apiVersionWithGroup: string;
+}
+
 export class KubeApi<T extends KubeObject = any> {
-  static matcher = /(\/apis?.*?)\/(?:(.*?)\/)?(v.*?)(?:\/namespaces\/(.+?))?\/([^\/]+)(?:\/([^\/?]+))?.*$/
+  static matcher = /(\/apis?.*?)\/(?:(.*?)\/)?(.*?)(?:\/namespaces\/(.+?))?\/([^\/]+)(?:\/([^\/?]+))?.*$/
 
-  static parseApi(apiPath = "") {
+  static parseApi(apiPath = ""): IKubeApiLinkBase {
     apiPath = new URL(apiPath, location.origin).pathname;
     const [, apiPrefix, apiGroup = "", apiVersion, namespace, resource, name] = apiPath.match(KubeApi.matcher) || [];
     const apiVersionWithGroup = [apiGroup, apiVersion].filter(v => v).join("/");
     const apiBase = [apiPrefix, apiGroup, apiVersion, resource].filter(v => v).join("/");
+
+    if (!apiBase) {
+      throw new Error(`invalid apiPath: ${apiPath}`)
```

Comment:
Did you test how the dashboard is handling this error?

Reply:
I did, the rest of the dashboard works even if this throws.

Reply:
Good!